app.ts

```diff
@@ -0,0 +1,163 @@
+﻿const host = "http://localhost:2050";
+let grid: Grid;
+const initRecord = 0;
+const rowLimit = 22;
+
+class Grid {
+    from: number;
+    to: number;
+    headers: string[] = [];
+
+    $row = $("<div />", {
+        class: 'row'
+    });
+
+    $header = $("<div />", {
+        class: 'header'
+    });
+
+    $column = $("<div />", {
+        class: 'column'
+    });
+
+    //sets initial grid values.
+    constructor(from: number, to: number) {
+        this.from = from;
+        this.to = to;
+
+        $.getJSON(host + "/columns", (data) => { })
+            .done((headers) => {
+                this.headers = headers;
+                this.pageInit();
+
+                //adds initial records to row objects to be added to grid on page load.
+                $.getJSON(host + "/records?from=" + this.from + "&to=" + this.to, (data) => { })
```

Comment:
This bit is repeated, rather move it out into a method so we have one instance of it to maintain